### DISCUSSION TITLE
Using Log4J2 via log4j2-jboss-logmanager isn't working?

### DISCUSSION THREAD

**Discussion by saturna** [quarkusio/quarkus#29318] [Q&A] [answered]
Hello, I'm having issues getting log4j2 logger working with Quarkus.

Some background:

- This is for an existing project which was using log4j2, this is why I wanted to make the adapter work
- This project is an AWS Lambda

To Reproduce:

- Followed [this guide](https://quarkus.io/guides/amazon-lambda) and created AWS Lambda archetype using:
`mvn archetype:generate \
       -DarchetypeGroupId=io.quarkus \
       -DarchetypeArtifactId=quarkus-amazon-lambda-archetype \
       -DarchetypeVersion=2.14.0.Final`

- Added Log4J2 JBoss adapter as per instructions [here](https://quarkus.io/guides/logging#what-about-apache-log4j):
`<dependency>
    <groupId>org.jboss.logmanager</groupId>
    <artifactId>log4j2-jboss-logmanager</artifactId> 
</dependency>`

- Added:
`private static final org.apache.logging.log4j.Logger LOGGER = org.apache.logging.log4j.LogManager.getLogger(MyClass.class);`
- Then tried:
`LOGGER.error("Hello from Log4J!");`

- Also tried built in Quarkus'

`LOG.error("JBOSS Logger!");`
and
`io.quarkus.logging.Log.error("Hello from Panache");`

None of the above produce any logging output UNLESS I remove the added `log4j2-jboss-logmanager` dependency.

It appears that internally QuarkusDelayedHandler is used and it does add messages to `logRecords` queue but this handler is never activated.
If `log4j2-jboss-logmanager` is removed, ConsoleHandler is being used instead and everything works.

Any help would be appreciated, I'm probably missing something simple.

**Comment by saturna** [comment, marked as answer]
Answered my own question. The issue was that I was trying to use my Class' handler directly instead of using `io.quarkus.amazon.lambda.runtime.QuarkusStreamHandler::handleRequest` when provisioning this Lambda.

**Reply by gsmet** [reply]
Thanks for taking the time to report back!

**Comment by saturna** [comment]
Looks like I missed the disclaimer saying: `Do not change the handler switch. This must be hardcoded to io.quarkus.amazon.lambda.runtime.QuarkusStreamHandler::handleRequest. This handler bootstraps Quarkus and wraps your actual handler so that injection can be performed.`
